src/TextFieldRich.jsx

```diff
@@ -3,234 +3,439 @@ var UAParser = require('ua-parser-js');
+		return {
+			autoGrow: true,
+			value: "",
+		};
+	},
+	
+	componentDidMount: function() {
+		var iframe = this.refs.rte.getDOMNode();
+		iframe.style.width = "100%";
+		
+		// If height has been passed then set the iframe height
+		if (this.props.height) {
+			iframe.style.height = height + 'px';
+		}
+				
+		// Make the contents of the iframe editable
+		var cls = this;
+		
+		/*
+		 * We need to delay the settings of the iframe document body because react only renders the iframe
+		 * So after it is mounted, the iframe body is then created.
+		 * If we are not going to use setTimeout, the body settings are still set but will not reflect in the iframe document body
+		 */
+		setTimeout(function() { 
+					cls._enableDesign(true);
+					cls.setValue(cls.props.value);
+				}, 1); 
+	},
+	
+	componentDidUpdate: function() {
+		// Re enable the content editable of the iframe document
+		var cls = this;
+		setTimeout(function() {
+					cls._enableDesign(true);
+				}, 1);
+		
+	},
+
+	render: function() {
+		return (<div className="chamel-text-field-rich" ><iframe ref="rte" src="about:blank" /></div>);
+	},
+	
+	/**
+	 * Public interface to send commands to the RTE
+	 *
+	 * @param {string} command The name of the RTE command to execute
+	 * @param {string} option 	The option when executing a certain command. e.g. changing the font/background colors
+	 * @public
+	 */
+	sendCommand: function (command, option) {
+		try {
+			var idoc = this._getIframeDoc();
+			var iwnd = this._getIframeWindow();
+		
+			iwnd.focus();
+		  	idoc.execCommand(command, false, option);
+			iwnd.focus();
+		} 
+		catch (e) {
+			throw new Error('Error while executing the ' + command + ' command');
+		}
+	},
+	
+	/**
+     * Sets the color of the current selection in the iframe document
+     *
+     * @param {string} type		Type of command to be executed. Either forecolor or backcolor
+     * @param {string} color	The color that was selected
+     * @public
+     */
+	setColor: function (type, color) {
+		this._setRange();
+		this.sendCommand(type, color);
+	},
+	
+	/**
+	 * Inserts the html string
+	 * 
+	 * @param {string} html		The string that will be inserted		
+	 * @public
+	 */
+	insertHtml: function(html) {
+		this._setRange();
+		
+		if (uaParser.getBrowser().name == "IE") {
+			this.sendCommand('paste', html);
+		}
+		else {
+			this.sendCommand('insertHtml', html);
+		}
+	},
+	
+	/**
+	 * Prompts the dialog box for user input
+	 * 
+	 * @param {string} path		The url path to be linked on text
+	 * @public
+	 */
+	insertLink: function(path) {
+		this._setRange();
+		this.sendCommand("unlink", null);
+		this.sendCommand("createlink", path);
+	},
+	
+	/**
+	 * Clear the value of the iframe document
+	 *
+	 * @public
+	 */
+	clearValue: function() {
+		this.setValue('');
+	},
+	
+
+	/**
+	 * Get the current value of the iframe document / code mirror editor
+	 * 
+	 * @public
+	 */
+	getValue: function() {
+		if (!this.isMounted()) {
+			return "zzzzz";
```

Comment:
Oh, sorry about this.. I forgot to remove the zzzz.. i was testing this yesterday.. :)